### PR TITLE
Function finder refactoring

### PR DESCRIPTION
**Pull request by Xeeynamo**
Merges some code between the PSX and the Saturn function finders. The only functional change is the PSX function finder now exports in markdown. This is to prepare the integration of #396 in the CI.

### CODE REVIEW

File: tools/function_finder/helpers.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    best_result = None
+    best_percent = 0
+
+    for result in scratches["results"]:
+        if not "name" in result:
+            continue
+        # seems to give approximate matches, skip these
+        if result["name"] != name:
+            continue
+        if result["platform"] != "saturn":
```
Comment (sozud) [inline tools/function_finder/helpers.py:25]: I think the platform needs to be a parameter?

File: tools/function_finder/function_finder_saturn.py [hunk 2]
```diff
@@ -176,22 +105,21 @@ def print_github_flavored_table(data):
...
-            obj["%"] = ""
+            row[4] = results[i]["link"]
+            row[5] = results[i]["percent"]
 
-        to_print.append(obj)
+        to_print.append(row)
 
-    print_github_flavored_table(to_print)
+    headers = ["Filename", "Function Name", "Length", "Branches", "WIP", "%"]
+    print(tabulate(to_print, headers=headers, tablefmt="github"))
```
Comment (sozud) [inline tools/function_finder/function_finder_saturn.py:125]: I thought I tried this exact thing and couldn't get it to display right on GitHub. Do you have a branch pushed with the result?

File: tools/function_finder/function_finder_psx.py [hunk 3]
```diff
@@ -176,8 +146,8 @@ def find_wip(o):
     # Update output with the results
     for i, o in enumerate(output):
         # keep the in-source results as definitive
-        if o[4] == "":
```
Comment (sozud) [inline tools/function_finder/function_finder_psx.py:179]: Is the (Scraped) part being handled?

File: tools/function_finder/helpers.py (hunk 1 above)
Comment (Xeeynamo) [reply tools/function_finder/helpers.py]: I am glad reviewing brings light to oversights like this. Fixed this and the other change to link the scraper result to the function list

File: tools/function_finder/function_finder_saturn.py (hunk 2 above)
Comment (Xeeynamo) [reply tools/function_finder/function_finder_saturn.py:125]: It seems to work just fine. I originally tested it with a subset. This is the full dump after running the tool from the latest `make extract_saturn`: https://gist.github.com/Xeeynamo/a41713c9b2a610a67c95199641a19394

File: tools/function_finder/function_finder_psx.py (hunk 3 above)
Comment (sozud) [reply tools/function_finder/function_finder_psx.py:179]: Based on the result here it looks like we aren't indicating whether it's an in-source DECOMP_ME_WIP or scraped anymore. I don't mind much either way, just thought I should point this out. https://gist.github.com/Xeeynamo/a41713c9b2a610a67c95199641a19394

**Review by sozud — APPROVED** [review]
Seems ok to me

**Comment by sozud** [comment]
I'm going to go ahead and merge this to reduce the PR queue, will fix any breakage that comes up.

**Pull request merged**